dbt_subprojects/dex/macros/models/_project/eulerswap_compatible_trades.sql

```diff
@@ -0,0 +1,219 @@
+{% macro eulerswap_compatible_trades(
+    blockchain = null
+    , project = null
+    , version = null
+    , eulerswapinstance_evt_swap = null
+    , eulerswap_pools_created = null
+    , univ4_PoolManager_evt_Swap = null 
+    , filter = null 
+    )
+%}
+
+with 
+
+uni_v4_trades as (
+    select 
+        evt_tx_hash as tx_hash 
+        , evt_block_time as block_time 
+        , evt_index
+        , evt_block_number as block_number 
+    from 
+    {{ univ4_PoolManager_evt_Swap }}
+    where 1=1
+    {%- if is_incremental() %}
+    and {{ incremental_predicate('evt_block_time') }}
+    {%- endif %}
+
+),
+
+dexs as (
+    select 
+        s.evt_block_time as block_time
+        , s.evt_block_number as block_number 
+        , ep.asset0 as token0 
+        , ep.asset1 as token1
+        , ep.pool as maker -- uniswap pool id 
+        , ep.fee
+        , ep.protocolFee
+        , s.contract_address as instance -- hook
+        , s.contract_address as project_contract_address -- rename for dex trades
+        , ep.eulerAccount 
+        , ep.factory_address
+        , ep.creation_block_time as pool_creation_time 
+        , s.evt_tx_hash as tx_hash
+        , s.evt_index as tx_index 
+    
+        -- EulerSwap Instance event from pool's perspective, so flipping the sign to be user's perspective
+        , case 
+            when s.amount0In != 0 then -1 * cast(s.amount0In as double)
+            else cast(s.amount0Out as double)
+          end as amount0 
+        , case 
+            when s.amount1In != 0 then -1 * cast(s.amount1In as double)
+            else cast(s.amount1Out as double)
+          end as amount1
+        , s.amount0In
+        , s.amount0Out
+        , s.amount1In
+        , s.amount1Out
+        , s.sender -- router 
+        , s.evt_index
+        , case 
+            when t.tx_hash is not null then 'uni_v4' 
+            else 'OG' 
+         end as source
```

Comment:
do you have a specific use case for `source` other than filtering out downstream in final query reading this CTE? if not, you can remove need for the column and just filter `where t.tx_hash is null` to ignore rows written as uniswap v4 already.

Reply:
i see in your downstream macro below, you rewrite logic for `source` anyway. maybe this macro can be simplified to only filter out where lookup to uni v4 is null

Reply:
updated